build/commands/lib/config.js

```diff
@@ -408,6 +408,21 @@ Config.prototype.buildArgs = function () {
     args.use_vaapi = true
   }
 
+  // Enable Page Graph only in desktop builds.
+  if (this.targetOS !== 'android' && this.targetOS !== 'ios') {
+    args.enable_brave_page_graph = true
```

Comment:
we should only be using args to either override upstream values or set values that we don't want to store in gn (like urls, keys, etc..). Normal defaults should go in gn config, but I'd like to keep this disabled by default for now in builds unless we add a feature flag around it that is disabled by default.

Reply:
> we should only be using args to either override upstream values or set values that we don't want to store in gn (like urls, keys, etc..). Normal defaults should go in gn config, but I'd like to keep this disabled by default for now in builds unless we add a feature flag around it that is disabled by default.

these exact gn args are forced here to have ability to check them via python-overridden scripts by looking at `out/<dir>/args.gn` file during build.

Other ways to check gn args in upstream python scripts will add more conflict than we need.
If you have some other solution in mind, feel free to share it.

Reply:
> I'd like to keep this disabled by default for now in builds unless we add a feature flag around it that is disabled by default.

There is a feature flag that is disabled by default. Without this flag all `CoreProbeSink::HasAgentsGlobal(CoreProbeSink::kPageGraph)` calls will return false and all `probe::` calls related to PageGraph will do nothing.

We need to have this arg enabled by default to have `PageGraph` codebase always in a good shape, ready to be used.

Reply:
I'm not a big fan of having to read it from args.gn and I'd like to find another way, but for now can you add comments explaining?